packages/realm-server/tests/realm-server-test.ts

```diff
@@ -190,6 +192,234 @@ module('Realm Server', function (hooks) {
+      let response = await request
+        .patch('/_permissions')
+        .set('Accept', 'application/vnd.api+json')
+        .set(
+          'Authorization',
+          `Bearer ${createJWT(testRealm, 'mary', [
+            'read',
+            'write',
+            'realm-owner',
+          ])}`,
+        )
+        .send({
+          data: {
+            id: testRealmHref,
+            type: 'permissions',
+            attributes: {
+              permissions: {
+                bob: [],
+              },
+            },
+          },
+        });
+
+      assert.strictEqual(response.status, 204, 'HTTP 204 status');
+      let permissions = await fetchUserPermissions(dbAdapter, testRealmURL);
+      assert.deepEqual(
+        permissions,
+        {
+          mary: ['read', 'write', 'realm-owner'],
+        },
+        'permissions are correct',
+      );
+    });
+
+    test('cannot remove realm-owner permissions from PATCH /_permissions', async function (assert) {
+      let response = await request
+        .patch('/_permissions')
+        .set('Accept', 'application/vnd.api+json')
+        .set(
+          'Authorization',
+          `Bearer ${createJWT(testRealm, 'mary', [
+            'read',
+            'write',
+            'realm-owner',
+          ])}`,
+        )
+        .send({
+          data: {
+            id: testRealmHref,
+            type: 'permissions',
+            attributes: {
+              permissions: {
+                mary: [],
+              },
+            },
+          },
+        });
+
+      assert.strictEqual(response.status, 400, 'HTTP 400 status');
+      let permissions = await fetchUserPermissions(dbAdapter, testRealmURL);
+      assert.deepEqual(
+        permissions,
+        {
+          mary: ['read', 'write', 'realm-owner'],
+          bob: ['read', 'write'],
+        },
+        'permissions are correct',
+      );
+    });
+
+    test('cannot add realm-owner permissions from PATCH /_permissions', async function (assert) {
+      let response = await request
+        .patch('/_permissions')
+        .set('Accept', 'application/vnd.api+json')
+        .set(
+          'Authorization',
+          `Bearer ${createJWT(testRealm, 'mary', [
+            'read',
+            'write',
+            'realm-owner',
+          ])}`,
+        )
+        .send({
+          data: {
+            id: testRealmHref,
+            type: 'permissions',
+            attributes: {
+              permissions: {
+                mango: ['realm-owner', 'write', 'read'],
+              },
+            },
+          },
+        });
+
+      assert.strictEqual(response.status, 400, 'HTTP 400 status');
+      let permissions = await fetchUserPermissions(dbAdapter, testRealmURL);
+      assert.deepEqual(
+        permissions,
+        {
+          mary: ['read', 'write', 'realm-owner'],
+          bob: ['read', 'write'],
+        },
+        'permissions are correct',
+      );
+    });
+  });
```

Comment:
Is it worth having a test that asserts on invalid permissions, like `data: { nothing: null }` or the like?

Reply:
Sure I can add that

Reply:
done in https://github.com/cardstack/boxel/pull/1748/commits/d3525775a9eaa1b4b04947c37870b8a344632534